kythe/java/com/google/devtools/kythe/extractors/java/JavaCompilationUnitExtractor.java

```diff
@@ -932,6 +932,23 @@ public static ClassLoader getPlatformClassLoader() {
     return completeOptions;
   }
 
+  private static JavaCompiler findJavaCompiler() {
+    JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+    if (compiler == null) {
+      // This is all a bit of a hack to be able to extract OpenJDK itself, which
+      // uses a bootstrap compiler and a lot of JDK options to compile itself.
+      // Notably, when using modules the system compiler is inhibited and the actual compiler
+      // resides in jdk.compiler.iterim.  Rather than hard-code this, just fall back to the first
+      // JavaCompiler we can find.
+      logger.atWarning().log("Unable to find system compiler, using first available.");
+      JavaCompilationUnitExtractor.class.getModule().addUses(JavaCompiler.class);
+      ServiceLoader<JavaCompiler> sl =
+          ServiceLoader.load(JavaCompiler.class, ClassLoader.getSystemClassLoader());
```

Comment:
Should this be `JavaCompilationUnitExtractor.class.getModule().getClassLoader()`?

Reply:
I honestly don't know, but it's an easy enough change to make.  It also occurs to me that I'll have to do any of the calls involving modules via reflection to avoid breaking on JDK8 :-(